app/lib/infrastructure/language_repository.dart

```diff
@@ -2,132 +2,176 @@ part of 'package:word_link/domain/controllers/language_controller.dart';
 
 class _LanguageRepository extends LanguageController {
   @override
-  CollectionObject getMostUsedWords(
-      {required LanguageEnum language, int numberOfWords = 5,}) {
-    final Map<String, String> wordMap = _getWordMapForLanguage(language);
-
-    final wordPairs = wordMap.entries
-        .take(numberOfWords)
-        .map((entry) => CardObject(name: entry.key, answer: entry.value))
-        .toList();
-    return CollectionObject(
-        name: LanguageEnum.polish.displayName, cardsTemp: wordPairs,);
+  Future<CollectionObject> getMostUsedWords({
+    required LanguageEnum sourceLanguage,
+    LanguageEnum targetLanguage = LanguageEnum.english,
+    int numberOfWords = 5,
+  }) async {
+    final List<String> words = getSourceLanguageWords(sourceLanguage);
+
+    final cardCollection = await createCollectionFromWords(
+        sourceLanguage, targetLanguage, words.sublist(0, numberOfWords));
```

Comment:
`words.sublist(0, numberOfWords)`
Can be replaced with 
`words.sublist(numberOfWords)`

And there is no need for the return in the next line.
Just return the result from `createCollectionFromWords(...`